pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -389,6 +389,12 @@ func (p *provider) AddDefaults(spec v1alpha1.MachineSpec) (v1alpha1.MachineSpec,
 	if rawConfig.AssignPublicIP == nil {
 		rawConfig.AssignPublicIP = aws.Bool(true)
 	}
+	if rawConfig.IsSpotInstance != nil && *rawConfig.IsSpotInstance {
+		if spec.Labels == nil {
+			spec.Labels = map[string]string{}
+		}
+		spec.Labels["k8s-app"] = "aws-node-termination-handler"
```

Comment:
probably it's better to use something with OUR namespace (`k8c.io/aws-spot` for example, or anything else under our namespace).

Reply:
The reason why actually added this, is due to the addon template. The addon had the same labels from upstream. If I change here then I need to change it also in KKP and keep propagate this change with each time we upgrade the addon

Reply:
Yeah, the label IS needed no doubt, it's just the labels itself that should correspond to what happening.